packages/integration-karma/test/template/attribute-hidden/index.spec.js

```diff
@@ -0,0 +1,29 @@
+import { createElement } from 'lwc';
+
+import Test from 'x/test';
+
+function generateTestCases(testElement) {
+    const itTitle = testElement.textContent;
+    const expectedValue = !!(testElement.dataset.expected || false);
+
+    it(itTitle, () => {
+        expect(testElement.hidden).toBe(expectedValue);
```

Comment:
IMO, the test should also verify the reflected attribute value.

Reply:
sure, i can do that, but is always be reflected according to the property, therefore is going to be either `""` or `null`. ([See your comment](https://github.com/salesforce/lwc/pull/1952/files/1eb3fbff8c2dc00fe75d0a1fd44f058fb862f665#r448408856))

Reply:
The attribute value is always `""` or `null` is a side effect that we are using props to set attributes values. The props/attribute reflection is not a bijective operation. If in the future we change the compiler output to use props instead of attributes there will be a visible change.